test/SpokePoolClient.fills.ts

```diff
@@ -254,4 +264,53 @@ describe("SpokePoolClient: Fills", function () {
       "Block numbers out of range"
     );
   });
+
+  it("Creates a fills with different repaymentChainId", async function () {
+    const targetDeposit1 = { ...depositTemplate, depositId: depositTemplate.depositId.add(1) };
+    const targetDeposit2 = { ...depositTemplate, depositId: depositTemplate.depositId.add(2) };
+
+    // Submit multiple fills at the same block:
+    const startBlock = await spokePool.provider.getBlockNumber();
+    await fillRelay(spokePool, targetDeposit1, relayer1, {
+      repaymentChainId: originChainId2,
+      repaymentAddress: EvmAddress.from(relayer1.address),
+    });
+    await fillRelay(spokePool, targetDeposit2, relayer2, {
+      repaymentChainId: destinationChainId,
+      repaymentAddress: EvmAddress.from(relayer1.address),
+    });
+    const svmAddress = SvmAddress.from(Keypair.generate().publicKey.toBase58());
+    await fillRelay(spokePool, { ...depositTemplate, depositId: depositTemplate.depositId.add(3) }, relayer2, {
+      repaymentChainId: CHAIN_IDs.SOLANA,
+      repaymentAddress: svmAddress,
+    });
+    await hre.network.provider.send("evm_mine");
+
+    let fill1 = await findFillEvent(spokePool, targetDeposit1, startBlock);
+    expect(fill1).to.not.be.undefined;
```

Comment:
V. minor, but `to.exist` should do the same thing and is simpler.

```suggestion
    expect(fill1).to.exist;
```